apps/web/src/components/templates/BlueprintModal.tsx

```diff
@@ -8,17 +8,34 @@ import { When } from '../utils/When';
 import { ActivePageEnum } from '../../pages/templates/editor/TemplateEditorPage';
 import { errorMessage } from '../../utils/notifications';
 import { useEffect, useState } from 'react';
+import { updateUserOnBoarding } from '../../api/user';
+import { IUserEntity } from '@novu/shared';
+import { useSegment } from '../../hooks/use-segment';
 
 export function BlueprintModal() {
   const theme = useMantineTheme();
   const navigate = useNavigate();
+  const segment = useSegment();
   const onClose = () => {
+    segment.track('Blueprint canceled', {
+      blueprintId: localStorage.getItem('blueprintId'),
+    });
     localStorage.removeItem('blueprintId');
     navigate('/templates', {
       replace: true,
     });
   };
 
+  const { mutateAsync: updateOnBoardingStatus } = useMutation<
+    IUserEntity,
+    { error: string; message: string; statusCode: number },
+    { showOnBoarding: boolean }
+  >(({ showOnBoarding }) => updateUserOnBoarding(showOnBoarding));
+
+  async function disableOnboarding() {
+    await updateOnBoardingStatus({ showOnBoarding: false });
+  }
```

Comment:
We should be coherent with the spelling to not be confusing.
Checking the right spelling in the dictionary I think we should change to `updateUserOnboarding`, `updateOnboardingStatus` and `showOnboarding`.

Reply:
.... copied it from another part of our code and I think the object property needs to be `showOnBoarding`

Reply:
Ouch, I have just seen that is a schema property also. That one we shouldn't touch it. Damn. 😞